samples/client/petstore/cpp-qt5/build-and-test.bash

```diff
@@ -1,42 +1,48 @@
 #!/bin/bash
 
 set -e
+# export RUN_VALGRIND_TESTS=TRUE
 
 mkdir -p build
 cd build
 
-qmake ../PetStore/PetStore.pro CONFIG+=debug
+cmake ..
 
 make
 
-valgrind --leak-check=full ./PetStore |& tee result.log || exit 1
-
-echo "Make sure the tests are launched:"
-testCount=$(cat result.log | grep 'Finished testing of' | wc -l)
-if [ $testCount == 3 ]
-then
-  echo "Ok"
-else
-  echo "The tests were not run!!!"
-  exit 1
-fi
-
-echo "Make sure the tests passed:"
-successCount=$(cat result.log | grep '0 failed' | wc -l)
-if [ $successCount == 3 ]
-then
-  echo "Ok"
+if [[ -z "${RUN_VALGRIND_TESTS}" ]]; then
+    echo "Running Qt5 Petsore Tests"
+    ./cpp-qt5-petstore
 else
-  echo "The tests failed!!!"
-  exit 1
+  echo "Running Qt5 Petsore Tests with Valgrind"
+  valgrind --leak-check=full ./cpp-qt5-petstore |& tee result.log || exit 1
```

Comment:
You could use `--error-exitcode=1` to make valgrind return 1 when there are any issues (memory leaks or access to uninitialized memory).
Just a suggestion, it's fine for me as it is :grin:

Reply:
I deactivated the valgrind tests, because this result check is not robust.
`# export RUN_VALGRIND_TESTS=TRUE`
Waiting eagerly for your PR. :)

Reply:
I'm blind :grinning: 

> Waiting eagerly for your PR. :)

One day :wink: